app/models/user.rb

```diff
@@ -1,4 +1,25 @@
 class User < ActiveRecord::Base
   has_many :bids
-  validates :email, email: true, allow_blank: true
+
+  validates :email, allow_blank: true, email: true
+  validate :validate_credit_card_form
```

Comment:
I might do this. I had looked at another lib and it was just a wrapper around URI.parse which doesn't do much. The only reason to not do this might be if we wanted to blacklist certain URLs, but I think we'd cross that bridge later with another validator.